internal/ingress/controller/controller.go

```diff
@@ -491,17 +491,6 @@ func (n *NGINXController) getBackendServers(ingresses []*ingress.Ingress) ([]*in
 					server.Hostname, ingKey)
 			}
 
-			if server.ProxySSL.CAFileName == "" {
```

Comment:
Why are you removing this?
Without it, you can overwrite the server certificate, if the annotation is used in two different Ingress rules for the same hostname.

Reply:
Exactly that is the purpose here. The intent is to move the proxy-ssl rules to location level instead of the current server level. So users can then define certificates on location (backend/service) level. I.e. the different locations under the same server can have different certificates. This is an ingress usage pattern in our cloud.